src/devices/Bmp280/Bmp280.cs

```diff
@@ -0,0 +1,592 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+//ported from https://github.com/adafruit/Adafruit_BMP280_Library/blob/master/Adafruit_BMP280.cpp
+
+using System;
+using System.Device.I2c;
+using System.Device.I2c.Drivers;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+
+namespace Iot.Device
+{
+    public class Bmp280 : IDisposable, IBmp280
+    {
+        private I2cDevice _i2cDevice;
+        private CommunicationProtocol protocol;
+        private bool initialised = false;
+        private Bmp280CalibrationData CalibrationData;
+        protected readonly byte Signature = 0x58;
+        private int TFine;
+
+        private enum CommunicationProtocol
+        {
+            Gpio,
+            Spi,
```

Comment:
NIT: I would only add the I2c Communication Protocol for now on this enum, and add the Spi and Gpio ones once we add support for them.